src/test/java/com/internship/socialnetwork/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,177 @@
+package com.internship.socialnetwork.service.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import com.internship.socialnetwork.exception.BadRequestException;
+import com.internship.socialnetwork.exception.NotFoundException;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import com.internship.socialnetwork.dto.UserDto;
+
+import com.internship.socialnetwork.model.User;
+import com.internship.socialnetwork.repository.UserRepository;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class UserServiceImplTest {
+
+    private final Long userId1 = 1L;
+
+    private final Long userId2 = 2L;
+
+    private final String username1 = "user1";
+
+    private final String username2 = "user2";
+
+    private final String name1 = "John";
+
+    private final String name2 = "Alice";
+
+    private  final String surname1 = "Doe";
+
+    private final String surname2 = "Johnson";
+
+    private final String email = "test@email.com";
+
+    @Mock
+    private UserRepository userRepository;
+
+    @InjectMocks
+    private UserServiceImpl userService;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void shouldGetUser() {
+        User user = new User();
```

Comment:
Pls use given, when, then, and comments for all tests, check with your colleagues for example. Also verify calls should be used after every test